libsolidity/analysis/TypeChecker.cpp

```diff
@@ -510,6 +510,24 @@ bool TypeChecker::visit(VariableDeclaration const& _variable)
 			m_errorReporter.typeError(_variable.location(), "Internal or recursive type is not allowed for public state variables.");
 	}
 
+	if (_variable.isStateVariable())
```

Comment:
Please extract to function.

Reply:
ok, will do ASAP.

p.s.: the above questions have been solved after a short chat with @Marenz - I'll adapt his tests.